pkg/apis/pipeline/v1beta1/taskrun_types.go

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2018 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// TaskRunSpec defines the desired state of TaskRun
+type TaskRunSpec struct {
+	TaskRef TaskRef       `json:"taskRef"`
+	Trigger Trigger       `json:"trigger"`
+	Inputs  TaskRunInputs `json:"inputs,omitempty"`
+	Outputs Outputs       `json:"outputs,omitempty"`
+	Results Results       `json:"results"`
+}
+
+// TaskRunInputs holds the input values that this task was invoked with.
+type TaskRunInputs struct {
+	Sources []Source `json:"sources"`
+	Params  []Param  `json:"params,omitempty"`
+}
```

Comment:
Would the Task keep track of the version of the `Source` so it can run with multiple commits for example? Currently the `Source` includes the commit, but its is only identified in the task by the name, which would mean we would need a different source name for each commit?

Reply:
I see the `taskrun` has the Source object, so that should cover it